### DISCUSSION TITLE
Spa Hosting

### DISCUSSION THREAD

**Discussion by mickdelaney** [damikun/trouble-training#19] [Q&A] [answered]
Hi Dalibor,

Is it possible to host the Spa on something like AWS Cloudfront as long as the domain is the same ? 

https://app.mynapp.com/spa
https://app.mynapp.com/identity
https://app.mynapp.com/bff 

Thank you.

**Comment by damikun** [comment, marked as answer]
Hi,

I dont know much about AWS stack since im Google cloud user (GCE) ... But i believe it is just app engine...

In example the SPA is served from BFF directly (just static files react app..) . Because i wanna reduce demo number of services... But normaly you serve it form standalone app or CDN..  SPA is just static files or in case server-side rendered it is another app... So yes you can put SPA to standalone app im also doing this normally...

From architecture all parts can be deployed as standalone and just link and set firewall rules between + CORS list must be defined... Also the domain is required since HTTPS.. Demo use dev-certificates for localhost but real app require domain name and certs.

(Just for info)
Currlently demo can have issues with build (Not sure if it works) There was Node update and some providers switched to new version it can have a problem with package restore.. But only frontend part maybe...

**Comment by mickdelaney** [comment]
Thanks, we're currently on Idsrv v4 with React app running on a CDN, and planning on switching to Duende + BFF. 
So this is quite a nice project. 
Are you offering it as training in your day job ?

**Reply by damikun** [reply]
Yes thx that was idea to help or show some way to do it...

This was released as opensource version for workshop i did in past (person-in-person) im not full time trainer and for Duende it is much better to contact their official trainings.. They are also updating internal APIs so for example BFF have today some deprecated methods that i use in training it was released a year ago... 

This is simplified architecture im using it was more about OpenTelemetry and monitoring what was in beta that days and today is released and becomes standard..
